.github/workflows/bump_submodules.yml

```diff
@@ -0,0 +1,46 @@
+name: Bump Submodule
+
+on:
+  schedule:
+    - cron: '0 0 * * 0,4'  # Run at 00:00 on Sunday and Thursday
+  pull_request:
+    branches: ["main"]
+    paths:
+      - ".github/workflows/bump_submodules.yml"
+
+jobs:
+  bump_submodule:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        submodules: true
+
+    - name: Bump torch-mlir submodule
+      run: |
+        cd third_party/torch-mlir
+        git fetch origin
+        git checkout origin/main
+        torch_mlir_commit=$(git rev-parse HEAD)
```

Comment:
nit: capitalize bash variable to make it stand out from the commands? `TORCH_MLIR_COMMIT`